tsuru/client/deploy.go

```diff
@@ -231,21 +239,56 @@ func (c *AppDeploy) Run(context *cmd.Context, client *cmd.Client) error {
 			return err
 		}
 	}
+	c.m.Lock()
 	resp, err := client.Do(request)
 	if err != nil {
+		c.m.Unlock()
 		return err
 	}
 	defer resp.Body.Close()
-	_, err = io.Copy(&respBody, resp.Body)
-	if err != nil {
-		return err
+	c.eventID = resp.Header.Get("X-Tsuru-Eventid")
+	c.m.Unlock()
+	scanner := bufio.NewScanner(resp.Body)
```

Comment:
Question: The scanner here is only needed so that we can pause the output while cancel is called, right? The `safeStdout` writer already has an internal mutex, could the same goal be accomplished by calling `safeStdout.mu.Lock()` in the Cancel function (after exposing this mutex to the function somehow)? This would allow the io.Copy to remain in place and prevent freezing the output for long lines, or some clever \r usage.

Reply:
👍